src/gui/qgsmaptoolcapture.cpp

```diff
@@ -396,7 +396,14 @@ int QgsMapToolCapture::fetchLayerPoint( const QgsPointLocator::Match &match, Qgs
       if ( !f.geometry().vertexIdFromVertexNr( match.vertexIndex(), vId ) )
         return 2;
 
-      layerPoint = f.geometry().constGet()->vertexAt( vId );
+      if ( match.hasEdge() )
+      {
+        layerPoint = match.interpolatedPoint();
```

Comment:
Why re-fetch the geometry here (inside interpolatedPoint), when you've already retrieved it by this stage?

Reply:
Since, it's retrieved in 2D only when match hasEdge. But maybe, I'm missing something - and that would explain why it works when digitinzing into the app and why some tests pass when they should fail. Did you see a better way to catch the Z value?

Reply:
> Since, it's retrieved in 2D only when match hasEdge.

I mean, you've got f.geometry() already. It's used in the not edge case. So it's rather inefficient to grab this again, when you could just do the interpolation on f.geometry() directly...

Reply:
Oh, yeah, right. I'll fix it. thanks :)